front/src/modules/rollingStock/components/rollingStockEditor/RollingStockEditorCurves.tsx

```diff
@@ -41,99 +43,142 @@ export default function RollingStockEditorCurves({
       selectedTractionMode !== THERMAL_TRACTION_IDENTIFIER &&
       currentRsEffortCurve.modes[selectedTractionMode]
-        ? currentRsEffortCurve.modes[selectedTractionMode].curves.reduce((acc, curve) => {
-            const rsElectricalProfile = curve.cond?.electrical_profile_level as string;
-            if (curve.cond?.comfort === selectedComfortLvl) {
-              acc.push(rsElectricalProfile);
-            }
-            return acc;
-          }, [] as string[])
+        ? currentRsEffortCurve.modes[selectedTractionMode].curves.reduce(
+            (acc, curve) => {
+              const rsElectricalProfile = curve.cond?.electrical_profile_level as string;
+              const rsPowerRestriction = curve.cond?.power_restriction_code as string;
+              if (curve.cond?.comfort === selectedComfortLvl) {
+                acc.push(rsElectricalProfile);
+                if (curve.cond?.electrical_profile_level === selectedElectricalProfile) {
+                  powerRestrictions.push(rsPowerRestriction);
+                }
+              }
+              return acc;
+            },
+            [null] as (string | null)[]
+          )
         : [];
-
     if (!rsComfortLevels.includes(selectedComfortLvl)) {
       dispatchComfortLvl(rsComfortLevels[0] || STANDARD_COMFORT_LEVEL);
     }
     if (selectedTractionMode === '' || !tractionModes.includes(selectedTractionMode)) {
       dispatchTractionMode(Object.keys(currentRsEffortCurve.modes)[0]);
-      dispatchElectricalProfil(rsElectricalProfiles[0]);
+      dispatchElectricalProfil(null);
+      dispatchPowerRestriction(null);
     }
     if (
-      (selectedElectricalProfile === null ||
-        !rsElectricalProfiles.includes(selectedElectricalProfile)) &&
-      !isEmpty(rsElectricalProfiles)
+      rsElectricalProfiles.length > 1 &&
+      !rsElectricalProfiles.includes(selectedElectricalProfile)
     ) {
       dispatchElectricalProfil(rsElectricalProfiles[0]);
     }
-    setRollingstockParams({
+    if (powerRestrictions.length > 1 && !powerRestrictions.includes(selectedPowerRestriction)) {
+      dispatchPowerRestriction(powerRestrictions[0]);
+    }
+
+    const rollingstockParamsLists = {
       comfortlevels: rsComfortLevels,
       tractionModes,
-      electricalProfiles: rsElectricalProfiles,
-    });
+      electricalProfiles: [...new Set(rsElectricalProfiles)],
+      powerRestrictions: [...new Set(powerRestrictions)],
+    };
+    setRollingstockParams(rollingstockParamsLists);
+    return rollingstockParamsLists;
   }, [selectedTractionMode, currentRsEffortCurve, selectedElectricalProfile]);
 
-  // **** We filter all the curves to find the one matching all selected params ****
-  const selectCurve = useCallback((): SelectedCurve => {
+  // **** We filter all the curves to find a list of curves matching the selected comfort and traction mode ****
+  const selectCurves = useCallback((): SelectedCurves => {
     if (!currentRsEffortCurve.modes[selectedTractionMode]) {
```

Comment:
Create a const and use it in the function
```suggestion
    const selectedTractionModeCurves = currentRsEffortCurve.modes[selectedTractionMode];
    if (!selectedTractionModeCurves) {
```